ssl/statem/statem_srvr.c

```diff
@@ -2148,7 +2252,13 @@ static int tls_early_post_process_client_hello(SSL_CONNECTION *s)
  err:
     sk_SSL_CIPHER_free(ciphers);
     sk_SSL_CIPHER_free(scsvs);
+#ifndef OPENSSL_NO_ECH
+    /* this doesn't seem to be an ECH specific change? */
+    if (clienthello != NULL)
+        OPENSSL_free(clienthello->pre_proc_exts);
```

Comment:
So clienthello can be NULL here only with ECH support? But even if so, I would just add the if() without the preprocessor guards.

Reply:
sure